src/EventSubscriber/RequestSubscriber.php

```diff
@@ -263,4 +274,35 @@ protected function constructQuerySpan($connection, array $query) {
     return $span;
   }
 
+  /**
+   * Add options to pass to the PHP Agents.
+   *
+   * @return array
+   *   An array of options to pass to the PHP Agent. Ie. tags.
+   */
+  protected function getRequestOptions() {
+    $options = [];
+
+    // Fetch the current path.
+    $route_object = $this->routeMatch->getRouteObject();
+    $path = $route_object->getPath();
+    $route_options = $route_object->getOptions();
+
+    // If this is an admin page, add a custom variable to denote that.
+    if (isset($route_options['_admin_route'])) {
+      $options['tags']['admin_page'] = TRUE;
+    }
+
+    // Add tags depending on the page we're in.
+    foreach (ELASTIC_APM_PARENT_ROUTES as $route) {
+      if (strpos($path, $route) === FALSE) {
+        continue;
+      }
+
+      $options['tags']['parent_route'] = $route;
+    }
```

Comment:
**CHANGE REQUEST**

Not sure how this works with the agent, but since you're passing this directly to the agent's constructor with further processing it I'd expect this to be a list of tags:

```
$options['tags'] = [
  'admin_page',
  'commerce',
];
```
while it seems it will be:
```
$options['tags'] = [
  'admin_page' => TRUE,
  'parent_route' => 'commerce',
];
```